Kafka 37 streams listing implementation

ivan-abc opened the pull request.

Commit:
Add example of Custom Kafka Headers

Commit:
Add headers handling, add options in yml files


"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]"

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Commit:
Add headers handling

oleg-rakitin committed:
Indentation correction

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]" 

Developer's Certificate of Origin 1.1 

By making a contribution to this project, I certify that: 

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or 

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or 

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it. 

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

oleg-rakitin committed:
Fix a forgotten bracket

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]" 

Developer's Certificate of Origin 1.1 

By making a contribution to this project, I certify that: 

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or 

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or 

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it. 

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

oleg-rakitin committed:
Update KafkaStorageDriver.java

oleg-rakitin committed:
Update KafkaStorageDriver.java

oleg-rakitin committed:
Fix errors

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]"

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

ivan-abc committed:
first version of list method implementetion

ivan-abc committed:
first version of list method implementetion

ivan-abc committed:
merge conflict resolved

ivan-abc committed:
Update KafkaStorageDriver.java

oleg-rakitin committed:
Merge branch 'master' into KAFKA-32-make-the-producer-record-key-and-headers-configurable

oleg-rakitin committed:
Add sharedHeaders

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]" 

Developer's Certificate of Origin 1.1 

By making a contribution to this project, I certify that: 

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or 

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or 

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it. 

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Commit:
Fix conflicts

Commit:
Fixes

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]"

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

oleg-rakitin committed:
Change the headers to the empty

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]"

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

oleg-rakitin committed:
Merge branch 'master' into KAFKA-32-make-the-producer-record-key-and-headers-configurable

ivan-abc committed:
spotless.eclipseformat.xml added

oleg-rakitin committed:
Change zlib version

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]" 

Developer's Certificate of Origin 1.1 

By making a contribution to this project, I certify that: 

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or 

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or 

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it. 

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

ivan-abc committed:
conflict resolved

ivan-abc committed:
gitlab-ci.yml changed

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -159,13 +166,27 @@ protected String requestNewAuthToken(Credential credential) {
+          buff.add(itemFactory.getItem(path + prefix + topicName, 0, 0));
+        }
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```
don't handle the InterruptedException instance, this may cause the hunging
use `throwUnchecked(e)` instead

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -159,13 +166,27 @@ protected String requestNewAuthToken(Credential credential) {
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      } catch (ExecutionException e) {
+        e.printStackTrace();
```
we use `LogUtil.exception(...)` instead

akurilov replied:
we use `LogUtil.exception(...)` instead

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -159,13 +166,27 @@ protected String requestNewAuthToken(Credential credential) {
-    buff.add(itemFactory.getItem(path + prefix, 0, 0));
+
+    if (itemFactory instanceof PathItemFactoryImpl) {
+      Properties properties = new Properties();
```
use type inference and final modifier where possible
`val properties = new Properties();`
or
`final var properties = new Properties();`

akurilov replied:
We have a properties instances cache:
https://github.com/emc-mongoose/mongoose-storage-driver-kafka/blob/966c7772279e857af5b076dd07669a318c4247ff/src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java#L52

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -159,13 +166,27 @@ protected String requestNewAuthToken(Credential credential) {
+
+    if (itemFactory instanceof PathItemFactoryImpl) {
+      Properties properties = new Properties();
+      adminClient = KafkaAdminClient.create(properties);
```
We have the admin client instance cache:
https://github.com/emc-mongoose/mongoose-storage-driver-kafka/blob/966c7772279e857af5b076dd07669a318c4247ff/src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java#L55

oleg-rakitin committed:
Merge branch 'master' into KAFKA-32-make-the-producer-record-key-and-headers-configurable

oleg-rakitin committed:
Check for empty default headers

"Signed-off-by: [Oleg Rakitin] [limonchik16@gmail.com]"

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

ivan-abc committed:
changes to list method applied

oleg-rakitin committed:
Merge pull request #29 from emc-mongoose/KAFKA-32-make-the-producer-record-key-and-headers-configurable

Add example of Custom Kafka Headers in options, add headers handling

Merk1998 committed:
Update README.md

Merk1998 committed:
Update README.md

Merk1998 committed:
Merge pull request #43 from emc-mongoose/KAFKA-47-Create-records-performance-comparison

Add notes in README.md about records performance comparison

Commit:
performance improvements backport from the pravega driver

Commit:
autoformatting changes

Commit:
autoformatting changes

Commit:
making producer thread local

akurilov committed:
Merge pull request #46 from emc-mongoose/issue_KAFKA-51_performance

performance improvements backport from the pravega driver

Commit:
v4.2.7

Commit:
v4.2.8

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -369,13 +373,27 @@ protected String requestNewAuthToken(Credential credential) {
-    buff.add(itemFactory.getItem(path + prefix, 0, 0));
+
+    if (itemFactory instanceof PathItemFactoryImpl) {
+      val properties = configCache.get("");
```
use 1st node address, nodeAddrs[0], for example

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -369,13 +373,27 @@ protected String requestNewAuthToken(Credential credential) {
+    if (itemFactory instanceof PathItemFactoryImpl) {
+      val properties = configCache.get("");
+      val adminClient = adminClientCache.get("");
+      ListTopicsResult result = adminClient.listTopics();
```
val

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -369,13 +373,27 @@ protected String requestNewAuthToken(Credential credential) {
+      val adminClient = adminClientCache.get("");
+      ListTopicsResult result = adminClient.listTopics();
+      try {
+        for (String topicName : result.names().get()) {
```
val?

ivan-abc committed:
required changes to list method were made

ivan-abc committed:
merge conflict fixed
